docs/dynamicproxy-async-interception.md

```diff
@@ -0,0 +1,290 @@
+# Asynchronous interception
+
+This article discusses several interception scenarios related to asynchrony. We'll look at these scenarios in increasing order of implementation difficulty:
+
+ * Intercepting awaitable methods that don't produce a return value (e.g. `Task`), with and without using `async`/`await` in interceptor code
+
+ * Intercepting awaitable methods that do produce a return value (e.g. `Task<TResult>`), with using `async`/`await` in interceptor code
+
+ * Using `invocation.Proceed()` in combination with `async`/`await`
+
+This article contains C# code examples that make use of the `async`/`await` keywords. Before you proceed, please make sure that you have a good understanding of how `async`/`await` works. See e.g. the article ["Asynchronous Programming with `async` and `await`" on docs.microsoft.com](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/).
```

Comment:
That MSDN page is pretty basic, there is a big gap between it and this article. I think adding a link to something that discusses how the compiler actually does things would be a good idea.

Maybe link to something like https://devblogs.microsoft.com/pfxteam/asyncawait-faq/, and make some brief comment that this DP article assumes that the developer is aware of the async/await machinery and that async/await is syntactic sugar over normal CLR methods.

Reply:
Oh, you're right, Stephen Toub's FAQ is a much better resource. I've updated this paragraph.